File: c/insert_case1.c
#include <whitedb/dbapi.h>
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int run_test(int nr);

int main(int argc, char **argv) {
    struct timespec start, stop;
    int i;

    for (i = 0; i < 10; i++) {
        clock_gettime(CLOCK_REALTIME, &start);
        run_test(i);
        clock_gettime(CLOCK_REALTIME, &stop);
        diff(start, stop);
    }
    return (EXIT_SUCCESS);
}


int run_test(int nr) {
    void *db;
    void *record;
    char *name = "1";
    int i, j;
    int field_count = 100;

    db = wg_attach_database(name, 57671680);  // 55 megabytes
    if (!db) {
        printf("failed at %d\n", nr);
        exit(0);
    }
    for (i = 0; i < 68000; i++) {
        record = wg_create_record(db, field_count);
        for (j = 0; j < field_count; j++) {
            wg_set_int_field(db, record, j, j);
        }
    }

    wg_detach_database(db);
    wg_delete_database(name);
    return 0;
}

File: python/insert_case1.py
import time
import wgdb

def runTest():
	field_count = 100
	db = wgdb.attach_database("1", 57671680);
	for x in range(0, 68000):
		record = wgdb.create_record(db, field_count)
		for j in range(0,field_count):
			wgdb.set_field(db, record, j, j)
	wgdb.detach_database(db);
	wgdb.delete_database("1");
	return;
	
def printDiff(start, end):
	diff = end - start;
	
	
for i in range(0,10):
	start = time.clock();
	runTest();
	print 'Time in function', time.clock() - start
	
